xbmc/settings/dialogs/GUIDialogAudioDSPManager.cpp

```diff
@@ -704,14 +706,24 @@ void CGUIDialogAudioDSPManager::Update()
     return;
   }
 
-  for (int iModeType = 0; iModeType < AE_DSP_MODE_TYPE_MAX; iModeType++)
+  // construct a CFileItemList to pass 'em on to the list
+  CFileItemList items;
+  for (int i = 0; i < AE_DSP_MODE_TYPE_MAX; ++i)
   {
+    int iModeType = dsp_mode_types[i].iModeType;
+
     modes.clear();
     db.GetModes(modes, iModeType);
 
     // No modes available, nothing to do.
     if (!modes.empty())
     {
+      CFileItemPtr item(new CFileItem());
+      item->SetLabel(g_localizeStrings.Get(dsp_mode_types[i].iName));
+      item->SetLabel2(g_localizeStrings.Get(dsp_mode_types[i].iDescription));
+      item->SetProperty("currentMode", dsp_mode_types[i].sModeType);
```

Comment:
Great idea @AlwinEsch, it's much better to do set this in our core codebase instead in every skin.